src/main/kotlin/org/rust/lang/core/types/visitors/impl/RustTypificationEngine.kt

```diff
@@ -217,7 +217,7 @@ private class RustTypeTypificationVisitor(val pivot: RsType) : RustComputingVisi
             RustUnitType
     }
 
-    override fun visitBaseType(o: RsBaseType) = set {
+    override fun visitBaseType(o: RsBaseType): Unit = o.baseType?.let { visitBaseType(it) } ?: set {
```

Comment:
This `ComputingVisitor`s are tricky. By convention, every method should should start with `= set`, to make sure that the result is set exactly once. Can you move this inside the `set`?